content/source/docs/enterprise/workspaces/run-basics.html.md

```diff
@@ -26,17 +26,79 @@ When you request a run, TFE locks the run to the current Terraform code (usually
+
+### 3. The Policy Check Stage
+
+This stage only occurs if [Sentinel policies][] are enabled. After a successful `terraform plan`, TFE checks whether the plan obeys policy to determine whether it can be applied.
+
+[Sentinel policies]: ../sentinel/index.html
+
+_States in this stage:_
+
+- **Policy Check:** TFE is currently checking the plan against the organization's policies.
+- **Policy Override:** The policy check finished, but a soft-mandatory policy failed, so an apply cannot proceed without approval from a member of the owners team. TFE pauses in this state.
+- **Policy Checked:** The policy check succeeded, and Sentinel will allow an apply to proceed. TFE sometimes pauses in this state, depending on workspace settings.
+
+_Leaving this stage:_
+
+- If any hard-mandatory policies failed, TFE skips to completion (**Plan Errored** state).
+- If any soft-mandatory policies failed, TFE pauses in the **Policy Override** state.
+    - If a member of the owners team overrides the failed policy, the run proceeds to the **Policy Checked** state.
+    - If an owner or a user with write access discards the run, TFE skips to completion (**Discarded** state).
+- If the run reaches the **Policy Checked** state (no mandatory policies failed, or soft-mandatory policies were overridden):
+    - If auto-apply is enabled on the workspace, TFE proceeds automatically to the apply stage.
+    - If auto-apply is disabled, TFE pauses in the **Policy Checked** state until a user with write access takes action. The run proceeds to the apply stage if they approve the apply, or skips to completion (**Discarded** state) if they reject the apply.
+
+
+### 4. The Apply Stage
+
+_States in this stage:_
+
 - **Applying:** TFE is currently running `terraform apply`.
-- **Applied:** TFE has finished applying. The next run can now proceed.
-- **Plan Errored:** The `terraform plan` command failed, and you won't be able to apply this run. This usually means you need to fix variables or code. The next run can now proceed.
-- **Discarded:** The `terraform plan` command succeeded, but a user chose not to apply it. The next run can now proceed.
 
-In the list of workspaces on TFE's main page, each workspace shows the state of its most recently queued run. So if any runs are pending, the workspace is in "pending," regardless of what it's currently running or waiting for.
+_Leaving this stage:_
+
+- When the apply finishes, TFE proceeds automatically to completion (**Applied** state).
+
+### 5. Completion
+
+A run is considered completed if it finishes applying, if the plan or policy check fails, or if a user chooses not to continue it. Once a run is completed, the next run in the queue can enter the plan stage.
+
+_States in this stage:_
+
+- **Applied:** TFE has finished applying.
+- **Plan Errored:** The `terraform plan` command failed (usually requiring fixes to variables or code), or a hard-mandatory Sentinel policy failed. The run cannot be applied.
```

Comment:
`terraform apply` command could have failed as well.

Reply:
About that! How can I learn which state (in the UI) it goes to if that happens? I don't know enough about Terraform to craft a configuration where the plan will succeed but the apply will fail, so I haven't been able to observe that yet.

Reply:
(...it's probably Plan Errored.)

Reply:
hmmmmm. someone with more terraform experience might be better suited to answer this...

oooo, you could have a plan that creates more resources than you are allowed in AWS. Like, try to create 10 VPCs in a region (I think the default limit is 5 VPCs). I would guess the plan would pass, but the apply would fail?

Reply:
![image](https://user-images.githubusercontent.com/484309/34844426-48529b96-f6c6-11e7-80cd-b927e10e5129.png)

Indeed, there's an Apply Errored state! And I found a way to do it myself: apply a configuration with resources from (some provider), then delete all those resources (and any provider config block) and apply the config that would delete those resources. TFE will fail at apply until you add a provider config block for that provider.